cmake/hpx.cmake

```diff
@@ -0,0 +1,21 @@
+set(HPX_NAME hpx)
+
+find_package(MPI REQUIRED)
+
+message(STATUS "Building ${HPX_NAME}")
+ExternalProject_Add(${HPX_NAME}
+ SOURCE_DIR ${PROJECT_SOURCE_DIR}/hpx
+ CONFIGURE_COMMAND ${CMAKE_COMMAND}
+   -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+   -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
+   -DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}
+   -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
+   -DCMAKE_C_FLAGS:STRING=${CMAKE_C_FLAGS}
+   -DMPI_C_COMPILER:FILEPATH=${MPI_C_COMPILER}
+   -DMPI_CXX_COMPILER:FILEPATH=${MPI_CXX_COMPILER}
+   -DCMAKE_SHARED_LINKER_FLAGS:STRING=${CMAKE_SHARED_LINKER_FLAGS}
+   -DBUILD_SHARED_LIBS:BOOL=${BUILD_SHARED_LIBS}
+   -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX}
+   -DHPX_WITH_MALLOC=system
+   <SOURCE_DIR>
+)
```

Comment:
We will need to add a `-DHPX_WITH_MPI=On` to the cmake command line in order for it to pick up the MPI environment. In terms of dependencies, `BOOST_ROOT` and `HWLOC_ROOT` might have to be set as well. I'm however not sure where this should go.

Reply:
Thanks, done!

Reply:
@junghans Sorry, now that I think about it a bit more, in the context of FleCSI we want for HPX not to do any networking at all, thus instead of the `-DHPX_WITH_MPI=On` we should use `-DHPX_WITH_NETWORKING=Off`.  Also we might want to disable building examples and tests, etc. This can be achieved by adding `-DHPX_WITH_EXAMPLES=Off` and `-DHPX_WITH_TESTS=Off`. Thanks!

Reply:
Done